src/shared/queue_linked_op.c

```diff
@@ -133,6 +156,10 @@ static void linked_queue_append_node(w_linked_queue_t *queue, w_linked_queue_nod
         queue->last = node;
     }
     queue->elements++;
+
+    if (queue->push_callback) {
+        queue->push_callback(node->data);
```

Comment:
In this case, I don't think that `data` could be NULL. But maybe we should also clarify in the header that the NULL check should be done in the callback.

Reply:
I added the warning as in the previous comment.